src/dmd/backend/cod1.d

```diff
@@ -1929,7 +1929,6 @@ void fixresult(ref CodeBuilder cdb, elem *e, regm_t retregs, regm_t *pretregs)
 
     if (sz == 1)
     {
-        assert(retregs & BYTEREGS);
```

Comment:
I need to be sure about the side effects of this CC @WalterBright .

Reply:
It does

Reply:
Now that I have a test case to cover it, I'm not sure if it's another side effect on `intel-intrinsics`. It seems to pass `test23084` successfully.